### PR TITLE
Removed the unused OAuth `/client-tokens` API

### LINKED ISSUE #4407: Remove the unused OAuth `/client-tokens` API
The OAuth server exposes some API endpoints under ` /client-tokens` that can be used for listing and deleting the OAuth clients attached to a user's account. We *used* to use these as part of the devices-and-apps list, but that has since switched over to using the new `/attached_clients` endpoint. Ripgrepping the code and inspecting request logs in bigquery, I don't think anything is using the `/client-tokens` endpoints any more.

We should remove them, which AFAICT would include:

* Deleting the `fetchOAuthApps` and `deleteOAuthApp` methods on the content-server's `oauth-client.js`.
* Removing the code for these routes in auth-server's `./lib/oauth/routes/client-tokens` and the plumbing that hooks them up.
* Removing the `getActiveClientsByUid` method from the auth-server's OAuth DB object, since it is only used by these routes.
* Removing auth-server's `lib/oauth/db/helpers.js` since it only contains a single function and that function is only used by `getActiveClientsByUid` above.
* Removing the `id` property from auth-server's `lib/oauth/db/accessToken.js`, since it only exists so these objects can be passed to the helper above.

Not a bad potential cleanup overall!

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-1286)
┆Issue Number: FXA-1286

### PR DESCRIPTION
**Pull request by imskr**
- Deleting the `fetchOAuthApps` and `deleteOAuthApp` methods on the content-server's oauth-client.js.

-  Removing the code for these routes in auth-server's `./lib/oauth/routes/client-tokens` and the plumbing that hooks them up.

- Removing the getActiveClientsByUid method from the auth-server's OAuth DB object, since it is only used by these routes.

- Removing auth-server's lib/oauth/db/helpers.js since it only contains a single function and that function is only used by getActiveClientsByUid above.

- Removing the id property from auth-server's lib/oauth/db/accessToken.js, since it only exists so these objects can be passed to the helper above.

Closes #4407 

@rfk

### CODE REVIEW

**Comment by imskr** [comment]
@rfk Build fails

**Comment by rfk** [comment]
Thanks for diving in here! From the test output I see this error message:

```
Error: Cannot find module '../../../lib/oauth/db/helpers'
```

Grepping around for references to that file, it looks like there is also a `packages/fxa-auth-server/test/oauth/db/helpers.js` file that can be deleted (since the code that it is testing has been deleted).

There are also some tests in the file `packages/fxa-auth-server/test/oauth/api.js` that will need to be deleted. I think most of them are in a block labelled `describe('/client-tokens'`, and since they're testing routes that no longer exist, they can be safely deleted.

**Comment by rfk** [comment]
OK great, looks like you're now seeing individual test failures from tests that are using the deleted code. If you click through to the CircleCI output view here:

* https://circleci.com/gh/mozilla/fxa/158288?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

It should show a listing of the failing tests. I took a quick look over them and from the look of things, we just need to delete a bunch of tests because they're testing code that no longer exists!

The CircleCI output includes file and line-number references that should help in tracking down the tests to be deleted, but let me know if you'd like any more guidance.

**Comment by imskr** [comment]
Thanks @rfk I will fix it

**Comment by imskr** [comment]
After a little struggle, all tests passed @rfk. Please Review!

File: packages/fxa-auth-server/test/oauth/db/index.js [hunk 1]
```diff
@@ -694,24 +679,13 @@ describe('db', function() {
...
-      });
-
       it('retrieves them with getAccessTokensByUid', async () => {
         const t = await db.generateAccessToken(tokenData);
         const tokens = await db.getAccessTokensByUid(userId);
         await db.removeAccessToken(t.tokenId);
         assert.isArray(tokens);
         assert.lengthOf(tokens, 1);
         assert.deepEqual(tokens[0].tokenId, t.tokenId);
-        assert.deepEqual(tokens[0].clientId, t.clientId);
```
[the remark this answers is not in the document]
Comment (imskr) [reply packages/fxa-auth-server/test/oauth/db/index.js]: It said Assertion Error

**Comment by imskr** [comment]
@rfk I've deleted the _getActiveClientsByUID from index.js and added the commit signing

**Comment by imskr** [comment]
Closing this. Will open new PR.

**Pull request closed without merging**